backend/nodes/utils/pil_utils.py

```diff
@@ -0,0 +1,110 @@
+from math import ceil, floor
+import os
+from typing import Tuple
+import cv2
+import numpy as np
+from sanic.log import logger
+
+
+class InterpolationMethod:
+    NEAREST = 0
+    LANCZOS = 1
+    LINEAR = 2
+    CUBIC = 3
+    BOX = 4
+
+
+try:
+    from PIL import Image, ImageDraw, ImageFont
+    from PIL.Image import Resampling
+
+    pil = Image
+
+    INTERPOLATION_METHODS_MAP = {
+        InterpolationMethod.NEAREST: Resampling.NEAREST,
+        InterpolationMethod.BOX: Resampling.BOX,
+        InterpolationMethod.LINEAR: Resampling.BILINEAR,
+        InterpolationMethod.CUBIC: Resampling.BICUBIC,
+        InterpolationMethod.LANCZOS: Resampling.LANCZOS,
+    }
+except ImportError:
+    logger.error("No PIL found, defaulting to cv2 for resizing")
+    Resampling = None
+    pil = Image = None
+    ImageDraw = None
+    ImageFont = None
+
+    INTERPOLATION_METHODS_MAP = {
+        InterpolationMethod.NEAREST: cv2.INTER_NEAREST,
+        InterpolationMethod.BOX: cv2.INTER_AREA,
+        InterpolationMethod.LINEAR: cv2.INTER_LINEAR,
+        InterpolationMethod.CUBIC: cv2.INTER_CUBIC,
+        InterpolationMethod.LANCZOS: cv2.INTER_LANCZOS4,
+    }
+
+
+def resize(
+    img: np.ndarray, out_dims: Tuple[int, int], interpolation: int
+) -> np.ndarray:
+    """Perform PIL resize or fall back to cv2"""
+
+    interpolation = INTERPOLATION_METHODS_MAP[interpolation]
+
+    # Try PIL first, otherwise fall back to cv2
+    if pil is not None:
+        pimg = pil.fromarray((img * 255).astype("uint8"))
```

Comment:
Also, I just noticed that this is a bug. This assumes that the `img` is normalized. Same for where you copied this line.

Could you please fix this by normalizing `img` at the start of the method?

Reply:
Did I not include normalize? *checks* Apparently not. I had. No idea what happened to it.

Reply:
The Resize nodes currently noramlize the img before passing it to this `resize`. I forgot the change that.